Tests/EncrypterDecrypterInitializationTests.swift

```diff
@@ -25,131 +25,192 @@
 @testable import JOSESwift
 
+// https://stackoverflow.com/a/38725560/5233456
+private func ~= <T: Equatable>(array: [T], value: T) -> Bool {
+    return array.contains(value)
+}
+
 class EncrypterDecrypterInitializationTests: RSACryptoTestCase {
+    let rsaKeyManagementModeAlgorithms: [KeyManagementAlgorithm] = [.RSA1_5, .RSAOAEP, .RSAOAEP256]
+    let aesKeyManagementModeAlgorithms: [KeyManagementAlgorithm] = [.A128KW, .A192KW, .A256KW]
 
     @available(*, deprecated)
-    func testEncrypterDeprecated1RSAInitialization() {
-        for algorithm in KeyManagementAlgorithm.allCases {
-            guard algorithm != .direct else {
-                continue
-            }
-
+    func testEncrypterDeprecated1Initialization() {
+        for algorithm in rsaKeyManagementModeAlgorithms {
             XCTAssertNotNil(
-                Encrypter(keyEncryptionAlgorithm: algorithm, keyEncryptionKey: publicKeyAlice2048!, contentEncyptionAlgorithm: .A256CBCHS512)
+                Encrypter(
+                    keyEncryptionAlgorithm: algorithm,
+                    keyEncryptionKey: publicKeyAlice2048!,
+                    contentEncyptionAlgorithm: .A256CBCHS512
+                )
             )
         }
     }
 
     @available(*, deprecated)
-    func testEncrypterDeprecated2RSAInitialization() {
-        for algorithm in KeyManagementAlgorithm.allCases {
-            guard algorithm != .direct else {
-                continue
-            }
-
+    func testEncrypterDeprecated2Initialization() {
+        for algorithm in rsaKeyManagementModeAlgorithms {
             XCTAssertNotNil(
-                Encrypter(keyEncryptionAlgorithm: algorithm, encryptionKey: publicKeyAlice2048!, contentEncyptionAlgorithm: .A256CBCHS512)
+                Encrypter(
+                    keyEncryptionAlgorithm: algorithm,
+                    encryptionKey: publicKeyAlice2048!,
+                    contentEncyptionAlgorithm: .A256CBCHS512
+                )
             )
         }
     }
 
-    func testEncrypterNewRSAInitialization() {
+    func testSuccessfulEncrypterInitialization() {
         for algorithm in KeyManagementAlgorithm.allCases {
-            guard algorithm != .direct else {
-                continue
+            switch algorithm {
+            case rsaKeyManagementModeAlgorithms:
+                XCTAssertNotNil(
+                    Encrypter(
+                        keyManagementAlgorithm: algorithm,
+                        contentEncryptionAlgorithm: .A128CBCHS256,
+                        encryptionKey: publicKeyAlice2048!
+                    )
+                )
+            case aesKeyManagementModeAlgorithms:
+                XCTAssertNotNil(
+                    Encrypter(
+                        keyManagementAlgorithm: algorithm,
+                        contentEncryptionAlgorithm: .A128CBCHS256,
+                        encryptionKey: Data()
+                    )
+                )
+            default:
+                // .direct
```

Comment:
Not a huge fan of twisting switch statements like this (they're really more like `if` statements in this case as the exhaustiveness guarantee isn't quite there.

Not a huge enough issue that I'd ask for a change (I get this saves a ton of boilerplate), but big enough for me to comment on.

Reply:
I agree
. Updated it to be more explicit about the `.direct` case.